components in React

NosatskyiK opened the pull request:
Acceptance Criteria for 1

- [x]  Mentor is added to pr
- [x]  AC copied to pr description
- [x]  Link to deployed version is added to pr
- [x]  The ProductsList component correctly receives product data as props.
- [x]  ProductCard component created and renders a single product.
- [x]  Header and Footer elements are present.
- [x]  ProductList displays right on mobile devices

Acceptance Criteria for 1*
 

- [x] The Header component has clickable elements that toggle the display between the About and ProductsList components.
- [ ]  State management is handled in App.tsx, with state determining which component to show.
- [ ]  LocalStorage added to save items in card after page realod


https://nosatskyik.github.io/fe-react-2024/

Commit:
components in React

Commit:
fixed bug

vadym-zinchenko-moc commented on src/components/cart/Cart.component.tsx:
```diff
@@ -0,0 +1,22 @@
+export const Cart = () => {
+    const [hasNotAdded, setAdded] = useState(false);
+
+    const toggleCart = () => {
```
it's extra method, you can use method `setAdded`

vadym-zinchenko-moc commented on src/components/cart/Cart.component.tsx:
```diff
@@ -0,0 +1,22 @@
+import React, { useState } from 'react';
```
why you import `React`?

vadym-zinchenko-moc commented on src/App.tsx:
```diff
@@ -1,14 +1,22 @@
+import React, { useState } from 'react';
```
why you import `React`?

vadym-zinchenko-moc commented on src/components/cart/Cart.component.tsx:
```diff
@@ -0,0 +1,22 @@
+import styles from './cart.module.css';
+
+export const Cart = () => {
+    const [hasNotAdded, setAdded] = useState(false);
```
Bad naming. You must have the same name method and variable just to add `set` 
example - `[count, setCount]`

vadym-zinchenko-moc commented on src/components/productList/ProductList.component.tsx:
```diff
@@ -0,0 +1,12 @@
+
+export const ProductList = () => (
+    <main className={styles.main}>
+        <ProductCard data={productsData} />
```
`The ProductsList component correctly receives product data as props`

it seems that your `ProductsList`  must receive  productsData like props.
and `ProductCard` must receive product.

you can name `productsData` just like `products`

better to name props like original name of props  - `products={products}`

vadym-zinchenko-moc commented on src/components/productList/ProductList.component.tsx:
```diff
@@ -0,0 +1,12 @@
+import styles from './productList.module.css';
+
+export const ProductList = () => (
+    <main className={styles.main}>
```
why do you use `<main>` tag here?

vadym-zinchenko-moc commented on src/components/productList/productList.module.css:
```diff
@@ -0,0 +1,15 @@
+.main {
```
why do you use `main` class here?

Commit:
fixed bugs

NosatskyiK commented:
- Removed the unnecessary toggleCart function and replaced it with an arrow function in onClick.
- Removed unnecessary React imports
- Changed the name for useState in the cart
- Changed the productList logic
- Changed the main tag to a section and gave it the productList class

vadym-zinchenko-moc commented on src/components/productCard/ProductCard.component.tsx:
```diff
@@ -0,0 +1,19 @@
+import styles from './productCard.module.css';
+
+export const ProductCard: React.FC<{ product: Product }> = ({ product }) => (
+    <>
```
extra Fragment

Commit:
removed the unnecessary fragment
